### PR TITLE
gh-84545: Clarify the 'extend' action documentation in argparse

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-84545

----
📚 Documentation preview 📚: https://cpython-previews--125870.org.readthedocs.build/

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
@savannahostrowski, could you, as a native speaker, please look at these changes?

File: Doc/library/argparse.rst [hunk 1]
```diff
@@ -741,6 +741,21 @@ how the command-line arguments should be handled. The supplied actions are:
     >>> parser.parse_args('--str --int'.split())
     Namespace(types=[<class 'str'>, <class 'int'>])
 
+* ``'extend'`` - This stores a list, and appends each item from the multi-value
+  argument list to the list.
+  The ``'extend'`` action is typically used with the nargs_ keyword argument
+  value ``'+'`` or ``'*'``;
+  note that when nargs_ is ``None`` (by default) or ``'?'``, separate
+  characters of the argument string will be appended to the list.
+  Example usage::
```
Comment (savannahostrowski) [inline Doc/library/argparse.rst]:
```suggestion
* ``'extend'`` - This stores a list and appends each item from the multi-value
  argument list to it.
  The ``'extend'`` action is typically used with the nargs_ keyword argument
  value ``'+'`` or ``'*'``.
  Note that when nargs_ is ``None`` (the default) or ``'?'``, each
  character of the argument string will be appended to the list.
  Example usage::
```

Comment (savannahostrowski) [reply Doc/library/argparse.rst]: Should we also move this phrase up or better integrate it into the intro paragraph before listing the actions available?

> Only actions that consume command-line arguments (e.g. ``'store'``,
``'append'`` or ``'extend'``) can be used with positional arguments.

IMO, it seems a little disjoint at the bottom of the section now.

**Review by savannahostrowski — CHANGES_REQUESTED** [review]
Sounds good - just a couple of tweaks!

File: Doc/library/argparse.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/library/argparse.rst]: As for "the list"/"it" -- in descriptions of 'append' and 'append_const' they repeat "the list". Is it okay to have such difference, or "the list" should be replaced with "it" in other places too? Or there wass a reason to repeat "the list"?

> IMO, it seems a little disjoint at the bottom of the section now.

How do you suggest to move and change it? It is not at the bottom at the section -- there is a lot of text and examples after it. It is not disjoined -- it immediately follows the list of actions. Did you mean something other?

Comment (savannahostrowski) [reply Doc/library/argparse.rst]: > As for "the list"/"it" -- in descriptions of 'append' and 'append_const' they repeat "the list". Is it okay to have such difference, or "the list" should be replaced with "it" in other places too? Or there wass a reason to repeat "the list"?

It reads a bit redundant to me, but we can keep it for consistency.

> How do you suggest to move and change it? It is not at the bottom at the section -- there is a lot of text and examples after it. It is not disjoined -- it immediately follows the list of actions. Did you mean something other?

Thinking about this more, I think it's fine as you have it now. I considered whether it'd be worth having this as a pre-amble before introducing the actions, but I'm not sure I love it either. You can disregard this for now  😄 !

**Comment by serhiy-storchaka** [comment]
Thank you for your review @savannahostrowski. My English is very bad and I appreciate any help.

**Pull request merged**